### ISSUE TITLE
Execute shutdown handler functionality in failing unit tests

### ISSUE THREAD

**Issue by cyrille-artho** [cyrille-artho/modbat#75]
This is a part of issue #55.
At the moment, some test cases have a truncated output.

1. The "diff" function in the should show that one stream is shorter than the other one.
Current output:
```
SimpleCounter:
SimpleCounter1
it1 =        1 3ba471c1785a0175
it2 =        1 3ba471c1785a0175
```

Desired output (note: only mismatching lines, or the lines preceding a mismatch upon truncation, should be shown):
```
SimpleCounter:
SimpleCounter1
it1 =        1 3ba471c1785a0175
it2 =        1 3ba471c1785a0175
it1: end of file
it2 = 
```
(In this case, it2 has an empty line.)

2. (Only) for failing tests, the shutdown handler should be invoked explicitly. At the point where the exception is caught, the test harness should do the following:
```
  case e: Exception => {
    Modbat.ShutdownHandler.run
    ret=1
  }
```

**Comment by AlexandreFournel** [comment]
Unfortunately, the shutdown handler. run didn't change anything ...

**Comment by cyrille-artho** [comment]
Fixed in PR #96
